### PR TITLE
feat: add doas support

### LINKED ISSUE #4530: Feat Req: add doas support
## Why do you want this feature?
Some users may not use sudo but doas
## Are you interested in submitting a PR for this?
y

Also: https://github.com/coder/code-server/issues/5599

### PR DESCRIPTION
**Pull request by unknowndevQwQ**
Some people may have some reason to drop sudo and switch to doas

Fix: #4530 
Also: https://github.com/coder/code-server/pull/5600

### CODE REVIEW

**Comment by unknowndevQwQ** [comment]
I have read the CLA Document and I hereby sign the CLA

File: install.sh [hunk 1]
```diff
@@ -539,7 +541,7 @@ sudo_sh_c() {
 		echoh
 		echoerr "This script needs to run the following command as root."
 		echoerr "  $*"
-		echoerr "Please install sudo or su."
+		echoerr "Please install doas, sudo or su."
```
Comment (kylecarbs) [inline install.sh:544]: `doas` is less popular than sudo, so let's list it last.

```suggestion
		echoerr "Please install sudo, su, or doas."
```

**Review by kylecarbs — APPROVED** [review]
Thanks for the contribution!

**Comment by ammario** [comment]
@unknowndevQwQ `git apply` this to pass CI.

```patch
diff --git a/.github/workflows/typos.toml b/.github/workflows/typos.toml
index 90b0ea1c..d10d53c9 100644
--- a/.github/workflows/typos.toml
+++ b/.github/workflows/typos.toml
@@ -5,6 +5,8 @@ IST = "IST"
 MacOS = "macOS"
 
 [default.extend-words]
+# do as sudo replacement
+doas=doas
 
 [files]
 extend-exclude = [
```

**Pull request merged**